File: void-lib/void-orm-lib/void-standard-orm-lib/void-base-orm/src/main/java/cn/toutatis/xvoid/orm/base/authentication/entity/RequestAuthEntity.java
package cn.toutatis.xvoid.orm.base.authentication.entity;

import cn.toutatis.xvoid.common.standard.AuthFields;
import cn.toutatis.xvoid.common.enums.AuthType;
import com.alibaba.fastjson.JSONObject;


/**
 * 请求登录JSON转换实体
 * @author Toutatis_Gc
 */
public class RequestAuthEntity {

    /**
     * 登录载荷
     */
    private final JSONObject rawInformation;

    /**
     * 登录账户
     */
    private final String account;

    /**
     * 登录类型
     */
    private final AuthType authType;

    /**
     * 请求sessionId
     */
    private final String sessionId;

    public RequestAuthEntity(JSONObject rawInformation) {
        this.rawInformation = rawInformation;
        this.account = rawInformation.getString(AuthFields.ACCOUNT);
        this.authType = rawInformation.getString(AuthFields.AUTH_TYPE) != null ? AuthType.valueOf(rawInformation.getString(AuthFields.AUTH_TYPE)) : null;
        this.sessionId = rawInformation.getString(AuthFields.XVOID_INTERNAL_ACTIVITY_AUTH_SESSION_KEY);
    }

    public JSONObject getRawInformation() {
        return rawInformation;
    }

    public String getAccount() {
        return account;
    }

    public AuthType getAuthType() {
        return authType;
    }

    public String getSessionId() {
        return sessionId;
    }
}

File: void-spring-branch/void-spring-singleton/src/main/kotlin/cn/toutatis/xvoid/spring/core/security/access/auth/RequestAuthEntity.kt
package cn.toutatis.xvoid.spring.core.security.access.auth

import cn.toutatis.xvoid.common.standard.AuthFields
import cn.toutatis.xvoid.spring.core.security.access.AuthType
import com.alibaba.fastjson.JSONObject

/**
 * Request auth entity 请求实体
 * 请求登录接口传回的信息是一个json字符串
 * 在认证阶段将字符串转为此类对象
 * 此对象包含原始信息和转换信息
 * @constructor Create empty Request auth entity
 */
data class RequestAuthEntity(val rawInformation:JSONObject) {

    val account:String = rawInformation.getString(AuthFields.ACCOUNT)

    val authType: AuthType? = rawInformation.getObject(AuthFields.AUTH_TYPE,AuthType::class.java)

    val sessionId: String = rawInformation.getString(AuthFields.XVOID_INTERNAL_ACTIVITY_AUTH_SESSION_KEY)

}